pkg/lib/machine-info.es6

```diff
@@ -191,3 +191,46 @@ export function udev_info(address) {
     }
     return pr;
 }
+
+const memoryRE = /^(\w+[ \w+]*):\s(.*)/;
+
+function parseMemoryInfo(text) {
+    var info = {};
+    text.split("\n\n").map(paragraph => {
+        let locator = null;
+        let props = {};
+
+        paragraph = paragraph.trim();
+        if (!paragraph)
+            return;
+
+        paragraph.split("\n").map(line => {
+            line = line.trim();
+            let match = line.match(memoryRE);
+            if (match)
+              props[match[1]] = match[2];
+        });
+
+        locator = props["Locator"];
+        if (locator)
+            info[locator] = props;
+    });
+    return info;
+}
+
+var memory_info_promises = { };
+
+export function memory_info(address) {
+    var pr = memory_info_promises[address];
+    var dfd;
+
+    if (!pr) {
+        dfd = cockpit.defer();
+        memory_info_promises[address] = pr = dfd.promise();
+        var memory_device = "17";
+        cockpit.spawn(["/usr/sbin/dmidecode", "-t", memory_device], { err: "message" })
```

Comment:
I actually meant `-t memory`, which avoids the hardcoded magical "17" number altogether.

This also needs an `environ: ["LC_ALL=C"]` to ensure that the key names don't get translated.